src/ankh/ankh.go

```diff
@@ -40,19 +40,57 @@ type Context struct {
 // AnkhConfig defines the shape of the ~/.ankh/config file used for global
 // configuration options
 type AnkhConfig struct {
-	CurrentContextName string             `yaml:"current-context"` // note the intentionally offset names here
-	CurrentContext     Context            `yaml:"-"`               // (private) filled in by code
-	Contexts           map[string]Context `yaml:"contexts"`
+	SupportedEnvironments []string           `yaml:"supported_environments"`
+	SupportedProfiles     []string           `yaml:"supported_profiles"`
+	CurrentContextName    string             `yaml:"current_context"`
+	CurrentContext        Context            `yaml:"-"` // private, filled in by init code. The `-` instructs the yaml lib to not look for this field
+	Contexts              map[string]Context `yaml:"contexts"`
 }
 
 // ValidateAndInit ensures the AnkhConfig is internally sane and populates
 // special fields if necessary.
 func (ankhConfig *AnkhConfig) ValidateAndInit() []error {
 	errors := []error{}
+
+	if ankhConfig.CurrentContextName == "" {
+		errors = append(errors, fmt.Errorf("missing or empty `current_context`"))
+	}
+
+	if ankhConfig.SupportedEnvironments == nil {
```

Comment:
should this check the length of the array too?

Reply:
Good call, it probably should!